### PR TITLE
TGP-2385: updated nirms validation

### PR DESCRIPTION
**Pull request by besscerule**

### CODE REVIEW

File: app/forms/NirmsNumberFormProvider.scala [hunk 1]
```diff
@@ -27,7 +26,24 @@ class NirmsNumberFormProvider @Inject() extends Mappings {
   def apply(): Form[String] =
     Form(
       "value" -> text("nirmsNumber.error.required")
-        .transform(removeWhitespace, identity[String])
+        .transform(removeHyphensAndSpaces, identity[String])
         .verifying(regexp(StringFieldRegex.nirmsRegex, "nirmsNumber.error.invalidFormat"))
+        .transform(addHyphens, identity[String])
     )
+
+  def removeHyphensAndSpaces: String => String = _.replaceAll(" ", "")
```
Comment (ChrisCookOC) [inline app/forms/NirmsNumberFormProvider.scala]: does this actually remove hyphens anymore?

Comment (mauro-fanton) [inline app/forms/NirmsNumberFormProvider.scala]: should this be still named removedWhiteSpace, as the function only remove spaces? Or it is the way round? the function should also remove hyphen (-)?

Comment (besscerule) [reply app/forms/NirmsNumberFormProvider.scala]: good spot!

Comment (besscerule) [reply app/forms/NirmsNumberFormProvider.scala]: nope - i changed it and i've changed the name too

File: app/forms/NirmsNumberFormProvider.scala [hunk 2]
```diff
@@ -27,7 +26,24 @@ class NirmsNumberFormProvider @Inject() extends Mappings {
...
-        .transform(removeWhitespace, identity[String])
+        .transform(removeSpaces, identity[String])
         .verifying(regexp(StringFieldRegex.nirmsRegex, "nirmsNumber.error.invalidFormat"))
+        .transform(addHyphens, identity[String])
     )
+
+  def removeSpaces: String => String = _.replaceAll(" ", "")
+  def addHyphens: String => String = { original =>
+    val length = 11
+    if (original.length == length) {
```
Comment (mauro-fanton) [inline app/forms/NirmsNumberFormProvider.scala:37]: just a tip, you may want to look into patch here: e.g. 

"abdet67867".patch(3, "-", 0).patch(5, "-", 0)
> val res3: String = abd-e-t67867

**Review by mauro-fanton — APPROVED** [review]

**Pull request merged**